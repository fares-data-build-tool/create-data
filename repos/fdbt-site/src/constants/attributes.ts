// Session Attributes

export const FARE_TYPE_ATTRIBUTE = 'fdbt-fare-type';

export const CARNET_FARE_TYPE_ATTRIBUTE = 'fdbt-carnet-fare-type';

export const PASSENGER_TYPE_ATTRIBUTE = 'fdbt-passenger-type';

export const DEFINE_PASSENGER_TYPE_ERRORS_ATTRIBUTE = 'fdbt-passenger-type-errors';

export const SERVICE_ATTRIBUTE = 'fdbt-service';

export const JOURNEY_ATTRIBUTE = 'fdbt-journey';

export const FARE_STAGES_ATTRIBUTE = 'fdbt-fare-stages';

export const STAGE_NAMES_ATTRIBUTE = 'fdbt-stage-names';

export const DURATION_VALID_ATTRIBUTE = 'fdbt-duration-valid';

export const PRODUCT_DETAILS_ATTRIBUTE = 'fdbt-product-details';

export const TICKET_REPRESENTATION_ATTRIBUTE = 'fdbt-ticket-representation';

export const CSV_UPLOAD_ATTRIBUTE = 'fdbt-csv-upload';

export const FARE_ZONE_ATTRIBUTE = 'fdbt-csv-zone-upload';

export const PERIOD_EXPIRY_ATTRIBUTE = 'fdbt-period-expiry';

export const SERVICE_LIST_ATTRIBUTE = 'fdbt-services';

export const INPUT_METHOD_ATTRIBUTE = 'fdbt-input-method';

export const NUMBER_OF_STAGES_ATTRIBUTE = 'fdbt-number-stages';

export const MATCHING_ATTRIBUTE = 'fdbt-matching';

export const INBOUND_MATCHING_ATTRIBUTE = 'fdbt-inbound-matching';

export const NUMBER_OF_PRODUCTS_ATTRIBUTE = 'fdbt-number-of-products';

export const MULTIPLE_PRODUCT_ATTRIBUTE = 'fdbt-multiple-product';

export const PRICE_ENTRY_ATTRIBUTE = 'fdbt-price-entry';

export const SOP_ATTRIBUTE = 'fdbt-sales-offer-package';

export const SOP_INFO_ATTRIBUTE = 'fdbt-sales-offer-package-info';

export const SALES_OFFER_PACKAGES_ATTRIBUTE = 'fdbt-select-sales-offer-packages';

export const GROUP_SIZE_ATTRIBUTE = 'fdbt-group-size';

export const GROUP_PASSENGER_TYPES_ATTRIBUTE = 'fdbt-group-passenger-types';

export const GROUP_PASSENGER_INFO_ATTRIBUTE = 'fdbt-group-passenger-info';

export const GROUP_DEFINITION_ATTRIBUTE = 'fdbt-group-definition';

export const TIME_RESTRICTIONS_DEFINITION_ATTRIBUTE = 'fdbt-time-restrictions-definition';

export const RETURN_VALIDITY_ATTRIBUTE = 'fdbt-return-validity';

export const PRODUCT_DATE_ATTRIBUTE = 'fdbt-product-date-attribute';

export const MULTIPLE_OPERATOR_ATTRIBUTE = 'fdbt-multiple-operator-attribute';

export const MULTIPLE_OPERATORS_SERVICES_ATTRIBUTE = 'fdbt-multiple-operators-services-attribute';

export const FULL_TIME_RESTRICTIONS_ATTRIBUTE = 'fdbt-full-time-restrictions-attribute';

export const TERM_TIME_ATTRIBUTE = 'fdbt-term-time';

export const SCHOOL_FARE_TYPE_ATTRIBUTE = 'fdbt-school-fare-type';

export const FORGOT_PASSWORD_ATTRIBUTE = 'fdbt-reset-password';

export const USER_ATTRIBUTE = 'fdbt-user';

export const OPERATOR_ATTRIBUTE = 'fdbt-operator';

export const TXC_SOURCE_ATTRIBUTE = 'fdbt-txc-source';

export const MULTI_OP_TXC_SOURCE_ATTRIBUTE = 'fdbt-multi-op-txc-source';

export const REUSE_OPERATOR_GROUP_ATTRIBUTE = 'fdbt-reuse-operator-group';

export const SAVE_OPERATOR_GROUP_ATTRIBUTE = 'fdbt-save-operator-group';

export const SAVED_PASSENGER_GROUPS_ATTRIBUTE = 'fdbt-saved-passenger-groups';
