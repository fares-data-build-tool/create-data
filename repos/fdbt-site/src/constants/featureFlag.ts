// const dev = 'dev' === process.env.STAGE;
const devOrTest = ['test', 'dev'].includes(process.env.STAGE ?? '');
console.log(devOrTest);

export const globalSettingsDeleteEnabled = devOrTest;
export const myFaresEnabled = devOrTest;
export const saveProductsEnabled = true;
export const exportEnabled = false;
export const masterStopListEnabled = false;
