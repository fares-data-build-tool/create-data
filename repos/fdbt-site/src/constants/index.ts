export const OPERATOR_COOKIE: string = 'fdbt-operator-cookie';

export const SERVICE_COOKIE: string = 'fdbt-service-cookie';

export const FARETYPE_COOKIE: string = 'fdbt-faretype-cookie';

export const FEEDBACK_LINK: string = 'mailto:fdbt@transportforthenorth.com';

export const GOVUK_LINK: string = 'https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/';

export const CSV_INPUT_METHOD_COOKIE: string = 'fdbt-csv-input-method-cookie';

export const MANUAL_INPUT_METHOD_COOKIE: string = 'fdbt-manual-input-method-cookie';

export const INT_MAP_INPUT_METHOD_COOKIE: string = 'fdbt-interactive-map-input-method-cookie';
