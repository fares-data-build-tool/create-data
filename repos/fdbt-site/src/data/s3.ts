import AWS from 'aws-sdk';

export interface FareStage {
    stageName: string;
    prices: {
        price: string;
        fareZones: string[];
    }[];
}

export interface UserFareStages {
    fareStages: FareStage[];
}

export interface UserFareZone {
    FareZoneName: string;
    NaptanCodes: string;
    AtcoCodes: string;
}

const getS3Client = (): AWS.S3 => {
    let options = {};

    if (process.env.NODE_ENV === 'development') {
        options = {
            s3ForcePathStyle: true,
            accessKeyId: 'S3RVER',
            secretAccessKey: 'S3RVER',
            endpoint: new AWS.Endpoint('http://localhost:4572'),
        };
    }

    return new AWS.S3(options);
};

const s3 = getS3Client();

export const getUserFareStages = async (uuid: string): Promise<UserFareStages> => {
    if (!process.env.USER_DATA_BUCKET_NAME) {
        throw new Error('Environment variable for validated bucket not set');
    }

    const params = {
        Bucket: process.env.USER_DATA_BUCKET_NAME,
        Key: `${uuid}.json`,
    };
    const response = await s3.getObject(params).promise();
    const dataAsString = response.Body?.toString('utf-8') ?? '';

    return JSON.parse(dataAsString);
};

export const putStringInS3 = async (
    bucketName: string,
    key: string,
    text: string,
    contentType: string,
): Promise<void> => {
    const request: AWS.S3.Types.PutObjectRequest = {
        Bucket: bucketName,
        Key: key,
        Body: Buffer.from(text, 'binary'),
        ContentType: contentType,
    };

    await s3.putObject(request).promise();
};

export const putDataInS3 = async (
    data: UserFareZone[] | UserFareStages | string,
    key: string,
    processed: boolean,
): Promise<void> => {
    let contentType = '';
    let bucketName = '';

    if (!process.env.USER_DATA_BUCKET_NAME || !process.env.RAW_USER_DATA_BUCKET_NAME) {
        throw new Error('Bucket name environment variables not set.');
    }

    if (processed) {
        bucketName = process.env.USER_DATA_BUCKET_NAME;
        contentType = 'application/json; charset=utf-8';
    } else {
        bucketName = process.env.RAW_USER_DATA_BUCKET_NAME;
        contentType = 'text/csv; charset=utf-8';
    }

    await putStringInS3(bucketName, key, JSON.stringify(data), contentType);
};
