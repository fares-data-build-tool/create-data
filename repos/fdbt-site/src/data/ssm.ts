import { GetParameterCommand, SSMClient } from '@aws-sdk/client-ssm';

const ssm = new SSMClient({ region: 'eu-west-2' });

export const getSsmValue = async (name: string, withDecryption?: boolean): Promise<string> => {
    const input = {
        Name: name,
        WithDecryption: withDecryption ? withDecryption : true,
    };
    const value = (await ssm.send(new GetParameterCommand(input))).Parameter?.Value;

    if (!value) {
        throw new Error(`Parameter ${name} was not found`);
    }

    return value;
};
