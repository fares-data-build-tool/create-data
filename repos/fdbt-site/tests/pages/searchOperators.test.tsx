import * as React from 'react';
import { shallow } from 'enzyme';
import { getMockContext } from '../testData/mockData';
import { getSearchOperators, OperatorNameType } from '../../src/data/auroradb';
import SearchOperators, { getServerSideProps } from '../../src/pages/searchOperators';
import { SEARCH_OPERATOR_ATTRIBUTE } from '../../src/constants';

jest.mock('../../src/data/auroradb');

describe('pages', () => {
    describe('searchOperator', () => {
        const mockOperators: OperatorNameType[] = [
            {
                nocCode: 'BLAC',
                operatorPublicName: 'Blackpool',
            },
            {
                nocCode: 'BLAC',
                operatorPublicName: 'Blackpool operator',
            },
        ];

        beforeEach(() => {
            (getSearchOperators as jest.Mock).mockImplementation(() => mockOperators);
        });

        afterEach(() => {
            jest.resetAllMocks();
        });

        it('should render the page correctly without search results', () => {
            const tree = shallow(
                <SearchOperators errors={[]} searchText="" operators={[]} csrfToken="" pageProps={[]} />,
            );

            expect(tree).toMatchSnapshot();
        });

        it('should render the page correctly with search results', () => {
            const tree = shallow(
                <SearchOperators
                    errors={[]}
                    searchText=""
                    operators={[{ nocCode: 'BLAC', operatorPublicName: 'Blackpool' }]}
                    csrfToken=""
                    pageProps={[]}
                />,
            );

            expect(tree).toMatchSnapshot();
        });

        it('should render an error when errors are passed through', () => {
            const tree = shallow(
                <SearchOperators
                    errors={[{ errorMessage: 'Search requires a minimum of three characters', id: 'searchText' }]}
                    searchText=""
                    operators={[]}
                    csrfToken=""
                    pageProps={[]}
                />,
            );

            expect(tree).toMatchSnapshot();
        });
    });

    describe('getServerSideProps', () => {
        it('should return expected props to the page when the page is first visited by the user', async () => {
            (getSearchOperators as jest.Mock).mockImplementation(() => []);
            const ctx = getMockContext({ session: { [SEARCH_OPERATOR_ATTRIBUTE]: { errors: [] } } });
            const result = await getServerSideProps(ctx);

            expect(result.props.errors.length).toBe(0);
            expect(result.props.operators.length).toBe(0);
        });

        // below test needs addressing as unsure what it's proving
        it('should throw an error if noc invalid', async () => {
            const ctx = getMockContext({
                cookies: { operator: null },
                body: null,
                uuid: {},
                mockWriteHeadFn: jest.fn(),
                mockEndFn: jest.fn(),
                isLoggedin: false,
            });
            await expect(getServerSideProps(ctx)).rejects.toThrow('invalid NOC set');
        });

        // we need many more tests around this area, as there are none that show it works or how it deals with different requests.

        // it('should remove the users operator from the returned results', async () => {
        //     const ctx = getMockContext({
        //         cookies: { operator: { operator: { operatorPublicName: 'Blackpool Transport' }, noc: 'TEST' } },
        //         body: null,
        //         uuid: {},
        //         mockWriteHeadFn: jest.fn(),
        //         mockEndFn: jest.fn(),
        //         isLoggedin: true,
        //         query: '/searchOperators?searchOperator=Black',
        //     });
        //     const forgotPasswordSubmitSpy = jest.spyOn(aurora, 'getSearchOperators');
        //     forgotPasswordSubmitSpy.mockImplementation().mockResolvedValue([
        //         {
        //             operatorPublicName: 'Blackburn Transport',
        //             nocCode: 'BLACKB',
        //         },
        //         {
        //             operatorPublicName: 'Blackpool Transport',
        //             nocCode: 'TEST',
        //         },
        //         {
        //             operatorPublicName: 'Blackwell Transport',
        //             nocCode: 'BLACKW',
        //         },
        //     ]);
        //     await expect(getServerSideProps(ctx)).resolves.toBe({
        //         props: { errors: [], operators: [], searchText: '' },
        //     });
        // });
    });
});
