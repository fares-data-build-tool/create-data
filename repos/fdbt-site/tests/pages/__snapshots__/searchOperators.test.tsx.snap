// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages searchOperator should render a selection error when the user tries to add operators without selecting any search results 1`] = `
<Component
  description="Search Operators page for the Fares Data Build Tool"
  title="Search Operators - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/searchOperators"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="operator-search"
          >
            Search for the operators that the ticket covers
          </h1>
        </legend>
        <label
          className="govuk-label"
          htmlFor="search-input"
        >
          Operator name
        </label>
        <input
          className="govuk-input govuk-!-width-three-quarters"
          id="search-input"
          name="searchText"
          type="text"
        />
        <input
          className="govuk-button govuk-!-margin-left-5"
          id="search-button"
          name="search"
          type="submit"
          value="Search"
        />
      </fieldset>
    </div>
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search-results"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend"
        >
          <h2
            className="govuk-body-l govuk-!-margin-bottom-0"
            id="operator-search-results"
          >
            Your search for '
            <strong>
              blac
            </strong>
            ' returned
            <strong>
               
              1
               result
            </strong>
          </h2>
        </legend>
        <FormElementWrapper
          errorClass=""
          errorId="add-operator-checkbox-0"
          errors={Array []}
        >
          <div
            className="govuk-checkboxes"
          >
            <p
              className="govuk-hint"
              id="operator-hint-text"
            >
              Select the operators results and click add operator(s). This data is taken from the Traveline National Dataset.
            </p>
            <div
              className="govuk-checkboxes__item"
              key="checkbox-item-Blackpool"
            >
              <input
                className="govuk-checkboxes__input"
                id="add-operator-checkbox-0"
                name="operatorsToAdd"
                type="checkbox"
                value="BLAC#Blackpool"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="add-operator-checkbox-0"
              >
                Blackpool
              </label>
            </div>
          </div>
        </FormElementWrapper>
        <div
          className="govuk-!-margin-top-7"
        >
          <input
            className="govuk-button govuk-button--secondary"
            id="add-operator-button"
            name="addOperators"
            type="submit"
            value="Add Operator(s)"
          />
        </div>
      </fieldset>
    </div>
  </CsrfForm>
</Component>
`;

exports[`pages searchOperator should render a selection error when the user tries to remove selected operators without making a selection 1`] = `
<Component
  description="Search Operators page for the Fares Data Build Tool"
  title="Search Operators - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/searchOperators"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="selected-operators"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="selected-operators"
          >
            Here's what you have added
          </h1>
        </legend>
        <div
          className="govuk-inset-text"
        >
          <FormElementWrapper
            errorClass=""
            errorId="remove-operator-checkbox-0"
            errors={Array []}
          >
            <div
              className="govuk-checkboxes"
            >
              <div
                className="govuk-checkboxes__item"
                key="WBTR"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-0"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="WBTR#Warrington's Own Buses"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-0"
                >
                  Warrington's Own Buses
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="BLAC"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-1"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="BLAC#Blackpool Transport"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-1"
                >
                  Blackpool Transport
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="IWBusCo"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-2"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="IWBusCo#IW Bus Co"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-2"
                >
                  IW Bus Co
                </label>
              </div>
            </div>
          </FormElementWrapper>
          <input
            className="govuk-button govuk-button--secondary govuk-!-margin-top-5"
            id="remove-operators-button"
            name="removeOperators"
            type="submit"
            value="Remove Operators"
          />
        </div>
      </fieldset>
    </div>
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend--m"
        >
          <h1
            className="govuk-fieldset__heading"
            id="operator-search"
          >
            Search for more operators that the ticket covers
          </h1>
        </legend>
        <label
          className="govuk-label"
          htmlFor="search-input"
        >
          Operator name
        </label>
        <input
          className="govuk-input govuk-!-width-three-quarters"
          id="search-input"
          name="searchText"
          type="text"
        />
        <input
          className="govuk-button govuk-!-margin-left-5"
          id="search-button"
          name="search"
          type="submit"
          value="Search"
        />
      </fieldset>
    </div>
    <div>
      <input
        className="govuk-button"
        id="continue-button"
        name="continueButtonClick"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</Component>
`;

exports[`pages searchOperator should render an input error when the user makes an unsucessful search 1`] = `
<Component
  description="Search Operators page for the Fares Data Build Tool"
  title="Search Operators - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/searchOperators"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={
        Array [
          Object {
            "errorMessage": "Search requires a minimum of three characters",
            "id": "search-input",
          },
        ]
      }
    />
    <div
      className="govuk-form-group govuk-form-group--error"
    >
      <fieldset
        aria-describedby="operator-search"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="operator-search"
          >
            Search for the operators that the ticket covers
          </h1>
        </legend>
        <span
          className="govuk-error-message"
          id="search-input-error"
        >
          <span
            className="govuk-visually-hidden"
          >
            Error: 
          </span>
          Search requires a minimum of three characters
        </span>
        <label
          className="govuk-label"
          htmlFor="search-input"
        >
          Operator name
        </label>
        <input
          className="govuk-input govuk-!-width-three-quarters govuk-input--error"
          id="search-input"
          name="searchText"
          type="text"
        />
        <input
          className="govuk-button govuk-!-margin-left-5"
          id="search-button"
          name="search"
          type="submit"
          value="Search"
        />
      </fieldset>
    </div>
  </CsrfForm>
</Component>
`;

exports[`pages searchOperator should render just the search input when the user first visits the page 1`] = `
<Component
  description="Search Operators page for the Fares Data Build Tool"
  title="Search Operators - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/searchOperators"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="operator-search"
          >
            Search for the operators that the ticket covers
          </h1>
        </legend>
        <label
          className="govuk-label"
          htmlFor="search-input"
        >
          Operator name
        </label>
        <input
          className="govuk-input govuk-!-width-three-quarters"
          id="search-input"
          name="searchText"
          type="text"
        />
        <input
          className="govuk-button govuk-!-margin-left-5"
          id="search-button"
          name="search"
          type="submit"
          value="Search"
        />
      </fieldset>
    </div>
  </CsrfForm>
</Component>
`;

exports[`pages searchOperator should render the search input and a list of selected operators when search results have been selected 1`] = `
<Component
  description="Search Operators page for the Fares Data Build Tool"
  title="Search Operators - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/searchOperators"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="selected-operators"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="selected-operators"
          >
            Here's what you have added
          </h1>
        </legend>
        <div
          className="govuk-inset-text"
        >
          <FormElementWrapper
            errorClass=""
            errorId="remove-operator-checkbox-0"
            errors={Array []}
          >
            <div
              className="govuk-checkboxes"
            >
              <div
                className="govuk-checkboxes__item"
                key="WBTR"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-0"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="WBTR#Warrington's Own Buses"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-0"
                >
                  Warrington's Own Buses
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="BLAC"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-1"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="BLAC#Blackpool Transport"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-1"
                >
                  Blackpool Transport
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="IWBusCo"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-2"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="IWBusCo#IW Bus Co"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-2"
                >
                  IW Bus Co
                </label>
              </div>
            </div>
          </FormElementWrapper>
          <input
            className="govuk-button govuk-button--secondary govuk-!-margin-top-5"
            id="remove-operators-button"
            name="removeOperators"
            type="submit"
            value="Remove Operators"
          />
        </div>
      </fieldset>
    </div>
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend--m"
        >
          <h1
            className="govuk-fieldset__heading"
            id="operator-search"
          >
            Search for more operators that the ticket covers
          </h1>
        </legend>
        <label
          className="govuk-label"
          htmlFor="search-input"
        >
          Operator name
        </label>
        <input
          className="govuk-input govuk-!-width-three-quarters"
          id="search-input"
          name="searchText"
          type="text"
        />
        <input
          className="govuk-button govuk-!-margin-left-5"
          id="search-button"
          name="search"
          type="submit"
          value="Search"
        />
      </fieldset>
    </div>
    <div>
      <input
        className="govuk-button"
        id="continue-button"
        name="continueButtonClick"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</Component>
`;

exports[`pages searchOperator should render the search input and search results when there is a successful search 1`] = `
<Component
  description="Search Operators page for the Fares Data Build Tool"
  title="Search Operators - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/searchOperators"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="operator-search"
          >
            Search for the operators that the ticket covers
          </h1>
        </legend>
        <label
          className="govuk-label"
          htmlFor="search-input"
        >
          Operator name
        </label>
        <input
          className="govuk-input govuk-!-width-three-quarters"
          id="search-input"
          name="searchText"
          type="text"
        />
        <input
          className="govuk-button govuk-!-margin-left-5"
          id="search-button"
          name="search"
          type="submit"
          value="Search"
        />
      </fieldset>
    </div>
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search-results"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend"
        >
          <h2
            className="govuk-body-l govuk-!-margin-bottom-0"
            id="operator-search-results"
          >
            Your search for '
            <strong>
              blac
            </strong>
            ' returned
            <strong>
               
              1
               result
            </strong>
          </h2>
        </legend>
        <FormElementWrapper
          errorClass=""
          errorId="add-operator-checkbox-0"
          errors={Array []}
        >
          <div
            className="govuk-checkboxes"
          >
            <p
              className="govuk-hint"
              id="operator-hint-text"
            >
              Select the operators results and click add operator(s). This data is taken from the Traveline National Dataset.
            </p>
            <div
              className="govuk-checkboxes__item"
              key="checkbox-item-Blackpool"
            >
              <input
                className="govuk-checkboxes__input"
                id="add-operator-checkbox-0"
                name="operatorsToAdd"
                type="checkbox"
                value="BLAC#Blackpool"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="add-operator-checkbox-0"
              >
                Blackpool
              </label>
            </div>
          </div>
        </FormElementWrapper>
        <div
          className="govuk-!-margin-top-7"
        >
          <input
            className="govuk-button govuk-button--secondary"
            id="add-operator-button"
            name="addOperators"
            type="submit"
            value="Add Operator(s)"
          />
        </div>
      </fieldset>
    </div>
  </CsrfForm>
</Component>
`;

exports[`pages searchOperator should render the search input, search results and a list of selected operators when an additional search is made 1`] = `
<Component
  description="Search Operators page for the Fares Data Build Tool"
  title="Search Operators - Fares Data Build Tool"
>
  <CsrfForm
    action="/api/searchOperators"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="selected-operators"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend govuk-fieldset__legend--l"
        >
          <h1
            className="govuk-fieldset__heading"
            id="selected-operators"
          >
            Here's what you have added
          </h1>
        </legend>
        <div
          className="govuk-inset-text"
        >
          <FormElementWrapper
            errorClass=""
            errorId="remove-operator-checkbox-0"
            errors={Array []}
          >
            <div
              className="govuk-checkboxes"
            >
              <div
                className="govuk-checkboxes__item"
                key="WBTR"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-0"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="WBTR#Warrington's Own Buses"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-0"
                >
                  Warrington's Own Buses
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="BLAC"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-1"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="BLAC#Blackpool Transport"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-1"
                >
                  Blackpool Transport
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="IWBusCo"
              >
                <input
                  className="govuk-checkboxes__input"
                  id="remove-operator-checkbox-2"
                  name="operatorsToRemove"
                  type="checkbox"
                  value="IWBusCo#IW Bus Co"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="remove-operator-checkbox-2"
                >
                  IW Bus Co
                </label>
              </div>
            </div>
          </FormElementWrapper>
          <input
            className="govuk-button govuk-button--secondary govuk-!-margin-top-5"
            id="remove-operators-button"
            name="removeOperators"
            type="submit"
            value="Remove Operators"
          />
        </div>
      </fieldset>
    </div>
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend--m"
        >
          <h1
            className="govuk-fieldset__heading"
            id="operator-search"
          >
            Search for more operators that the ticket covers
          </h1>
        </legend>
        <label
          className="govuk-label"
          htmlFor="search-input"
        >
          Operator name
        </label>
        <input
          className="govuk-input govuk-!-width-three-quarters"
          id="search-input"
          name="searchText"
          type="text"
        />
        <input
          className="govuk-button govuk-!-margin-left-5"
          id="search-button"
          name="search"
          type="submit"
          value="Search"
        />
      </fieldset>
    </div>
    <div
      className="govuk-form-group "
    >
      <fieldset
        aria-describedby="operator-search-results"
        className="govuk-fieldset"
      >
        <legend
          className="govuk-fieldset__legend"
        >
          <h2
            className="govuk-body-l govuk-!-margin-bottom-0"
            id="operator-search-results"
          >
            Your search for '
            <strong>
              warri
            </strong>
            ' returned
            <strong>
               
              1
               result
            </strong>
          </h2>
        </legend>
        <FormElementWrapper
          errorClass=""
          errorId="add-operator-checkbox-0"
          errors={Array []}
        >
          <div
            className="govuk-checkboxes"
          >
            <p
              className="govuk-hint"
              id="operator-hint-text"
            >
              Select the operators results and click add operator(s). This data is taken from the Traveline National Dataset.
            </p>
            <div
              className="govuk-checkboxes__item"
              key="checkbox-item-Warrington's Own Buses"
            >
              <input
                className="govuk-checkboxes__input"
                id="add-operator-checkbox-0"
                name="operatorsToAdd"
                type="checkbox"
                value="WBTR#Warrington's Own Buses"
              />
              <label
                className="govuk-label govuk-checkboxes__label"
                htmlFor="add-operator-checkbox-0"
              >
                Warrington's Own Buses
              </label>
            </div>
          </div>
        </FormElementWrapper>
        <div
          className="govuk-!-margin-top-7"
        >
          <input
            className="govuk-button govuk-button--secondary"
            id="add-operator-button"
            name="addOperators"
            type="submit"
            value="Add Operator(s)"
          />
        </div>
      </fieldset>
    </div>
    <div>
      <input
        className="govuk-button"
        id="continue-button"
        name="continueButtonClick"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</Component>
`;
